File: Single/Easy/Skynet.the.Chasm.java
// Skynet: the Chasm
import java.util.*;

class Player {
    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        int R = in.nextInt();       // the length of the road before the gap.
        int G = in.nextInt();       // the length of the gap.
        int L = in.nextInt();       // the length of the landing platform.
        boolean beforeJump = true;
        int step = 0;
        // game loop
        while (true) {
            int S = in.nextInt();   // the motorbike's speed.
            int X = in.nextInt();   // the position on the road of the motorbike.
            System.err.println("gap:" + G + ", length:" + R + ", pos: " + X + ", speed:" + S);
            if (beforeJump) {
                if ((X+S) <= R) {
                    if (S < (G+1)) {
                        System.out.println("SPEED");
                    } else if (S > (G+1)) {
                        System.out.println("SLOW");
                    } else {
                        System.out.println("WAIT");
                    }
                } else {
                    System.out.println("JUMP");
                    beforeJump = false;
                }
            } else {
                System.out.println(S > 0 ? "SLOW" : "WAIT");
            }
            step++;
        }
    }
}

File: Single/Easy/Skynet.the.Chasm.php
<?php
// Skynet: the Chasm
// Output: one of 4 keywords: SPEED,SLOW,JUMP,WAIT
    fscanf(STDIN, "%d", $R);        // Length of road before gap
    fscanf(STDIN, "%d", $G);        // Length of gap
    fscanf(STDIN, "%d", $L);        // Length of landing platform
    $beforeJump = true;
    $step = 0;
    while (TRUE) {
        fscanf(STDIN, "%d", $S);    // Motorbike's speed
        fscanf(STDIN, "%d", $X);    // Position on road
        error_log("gap:" + $G + ", length:" + $R + ", pos: " + $X + ", speed:" + $S);
        if ($beforeJump) {
            if (($X + $S) <= $R) {
                if ($S < ($G + 1)) {
                    echo("SPEED\n");
                } else if ($S > ($G + 1)) {
                    echo("SLOW\n");
                } else {
                    echo("WAIT\n");
                }
            } else {
                echo("JUMP\n");
                $beforeJump = false;
            }
        } else {
            echo($S > 0 ? "SLOW\n" : "WAIT\n");
        }
        $step++;
    }
?>